### PR TITLE
Add parseOptions object to FastifyCookieOptions

### PR DESCRIPTION
**Pull request by MoSheikh**
Add `parseOptions` type definitions to `FastifyCookieOptions` interface and utilize options in `setCookie` function.

#### Checklist

- [X] run `npm run test` and `npm run benchmark`
- [X] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [X] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by darkgl0w — APPROVED** [review]
LGTM

**Review by zekth — APPROVED** [review]
✌️

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by MoSheikh** [comment]
Sorry, made a small change to clean the test up

**Review by zekth — APPROVED** [review]

**Comment by MoSheikh** [comment]
So I took another look and realized that the `setCookie` function that is added to the fastify instance wasn't doing anything with the options, so adding in an option such as `domain` or `path` as part of the plugin options on registration wouldn't do anything to the response. 

I made some changes to make the options propagate to the setCookie function. Please let me know if there are any changes you'd like me to make.

File: plugin.js [hunk 1]
```diff
@@ -80,8 +80,10 @@ function plugin (fastify, options, next) {
     return signer.unsign(value)
   }
 
-  function setCookie (name, value, options) {
-    return fastifyCookieSetCookie(this, name, value, options, signer)
+  function setCookie (name, value, cookieOptions) {
+    const opts = Object.assign({}, options.parseOptions || {})
+    Object.assign(opts, cookieOptions)
```
Comment (Eomm) [inline plugin.js:85]:
```suggestion
    const opts = Object.assign({}, options.parseOptions, cookieOptions)
```

Comment (MoSheikh) [reply plugin.js]: Thank you!

**Review by Eomm — APPROVED** [review]

**Review by zekth — APPROVED** [review]

**Review by darkgl0w — APPROVED** [review]

**Pull request merged**